lib/assessment.js

```diff
@@ -29,7 +45,11 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  array2 = [];
```

Comment:
This is assigning to a variable `array2` found outside this function's scope or creating a global variable called `array2`.